python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/asset_condition_tests/test_backfills_requested.py

```diff
@@ -0,0 +1,287 @@
+        state = state.with_current_time_advanced(hours=1)
+        state, new_run_requests = state.evaluate_tick_daemon()
+        assert len(new_run_requests) == 0
+
+        # parent gets materialized, B requested
+        state = state.with_runs(run_request("A", "2020-02-02-01:00"))
+        state, new_run_requests = state.evaluate_tick_daemon()
+        assert len(new_run_requests) == 1
+        # but it fails
+        state = state.with_failed_run_for_asset("B", "2020-02-02-01:00")
+
+        # B does not get immediately requested again
+        state, new_run_requests = state.evaluate_tick_daemon()
+        assert len(new_run_requests) == 0
+
+
+@mock.patch(
+    "dagster_tests.definitions_tests.auto_materialize_tests.test_asset_daemon.DagsterInstance.da_request_backfills",
+    return_value=True,
+)
+def test_disconnected_graphs_backfill(mock_da_request_backfills) -> None:
+    with get_daemon_instance() as instance:
+        state = (
+            AssetDaemonScenarioState(
+                two_disconnected_graphs.with_asset_properties(
+                    keys=["B", "D"],
+                    automation_condition=AutomationCondition.any_deps_match(
+                        AutomationCondition.newly_updated()
+                    ),
+                ),
+                request_backfills=True,
+                instance=instance,
+            )
+            .with_asset_properties(keys=["A", "B"], partitions_def=daily_partitions_def)
+            .with_asset_properties(keys=["C", "D"], partitions_def=two_partitions_def)
+            .with_current_time("2020-02-02T01:05:00")
+        )
+
+        # parent hasn't updated yet
+        state, new_run_requests = state.evaluate_tick_daemon()
+        assert len(new_run_requests) == 0
+
+        # historical parent updated, doesn't matter
+        # state = state.with_runs(run_request("A", "2019-07-05"))
+        # state, new_run_requests = state.evaluate_tick_daemon()
+        # assert len(new_run_requests) == 0
+
+        # A updated, now can execute B, but not D
+        state = state.with_runs(run_request("A", "2020-02-01"))
+        state, new_run_requests = state.evaluate_tick_daemon()
+        assert len(new_run_requests) == 1
+        assert new_run_requests[0].requires_backfill_daemon()
+        assert new_run_requests[0].asset_graph_subset
+        assert new_run_requests[0].asset_graph_subset.asset_keys == {AssetKey("B")}
+        state = state.with_runs(
+            *(
+                run_request(ak, pk)
+                for ak, pk in new_run_requests[0].asset_graph_subset.iterate_asset_partitions()
+            )
+        )
+
+        # now B has been materialized, so don't execute again
+        state, new_run_requests = state.evaluate_tick_daemon()
+        assert len(new_run_requests) == 0
+
+        # new partition comes into being, parent hasn't been materialized yet
+        state = state.with_current_time_advanced(days=1)
```

Comment:
I think overall, this sort of test was targeted more at the specific properties of the `AutoMaterializePolicy.eager()`, which tried to materialize any missing partition of an asset. Because this is operating on a simpler policy, I don't think this bit is all that relevant.

That being said, I think the interesting tests will come once the `will_be_requested` condition logic is updated to allow backfills to chain together multiple different partitions definitions (as that's where we'll actually see behavior differences), so maybe we can take a pass at making these tests more tailored at that point in time.

Reply:
yeah I agree that some of these tests dont add much. I mainly wanted to get a couple different scenarios written up to  build off of when will_be_requested gets updated